### PR TITLE
CY-1905 Auto configure manager status reporter

### PR DESCRIPTION
**Pull request by davidginzbourg**

### CODE REVIEW

File: tests/integration_tests/tests/agentless_tests/test_status_reporter.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        snapshot_id = "test_" + str(uuid4())
+        initial_token_key, _ = self.get_reporter_token_key()
+
+        execution = self.client.snapshots.create(snapshot_id, False)
+        self.wait_for_execution_to_end(execution)
+
+        self.update_reporter_token_key(uuid4().hex)
+
+        execution = self.client.snapshots.restore(snapshot_id)
+        # Temporary fix until CY-1821 is fixed.
```
Comment (tyacbovi) [inline tests/integration_tests/tests/agentless_tests/test_status_reporter.py]: I would rather to open a jira about this refactor

File: tests/integration_tests/tests/agentless_tests/test_status_reporter.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+class TestManagerStatusReporter(AgentlessTestCase):
+    def test_status_reporter_has_correct_key_after_snapshot_restore(self):
+        snapshot_id = "test_" + str(uuid4())
+        initial_token_key, _ = self.get_reporter_token_key()
+
+        execution = self.client.snapshots.create(snapshot_id, False)
+        self.wait_for_execution_to_end(execution)
+
+        self.update_reporter_token_key(uuid4().hex)
```
Comment (tyacbovi) [inline tests/integration_tests/tests/agentless_tests/test_status_reporter.py]: you can also just use a const string and not uuid4

File: tests/integration_tests/tests/agentless_tests/test_status_reporter.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        # Temporary fix until CY-1821 is fixed.
+        sleep(40)
+        #  At this point it should be safe to query executions.
+        self.wait_for_execution_to_end(execution)
+
+        current_token_key, reporter_id = self.get_reporter_token_key()
+        self.assertEqual(initial_token_key, current_token_key)
+
+        reporter_configuration_json = self.execute_on_manager(
+            "cfy_manager status-reporter show-configuration --json").stdout
```
Comment (tyacbovi) [inline tests/integration_tests/tests/agentless_tests/test_status_reporter.py]: I will separate it to func so the test steps will be clearer

File: tests/integration_tests/tests/agentless_tests/test_status_reporter.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            ''.format(MANAGER_PYTHON, script_dst, reporter_id)).stdout.strip()
+        full_token = '{0}{1}'.format(encoded_id, initial_token_key)
+        self.assertEqual(reporter_configuration['token'], full_token)
+
+        reporter_service_status = self.execute_on_manager(
+            "sh -c 'systemctl is-active cloudify-status-reporter || :'"
+        ).stdout.strip()
+        self.assertEqual(reporter_service_status, 'active')
+
+    def get_reporter_token_key(self):
```
Comment (tyacbovi) [inline tests/integration_tests/tests/agentless_tests/test_status_reporter.py]: the next funcs are more internal

File: tests/integration_tests/tests/agentless_tests/test_status_reporter.py (hunk 2 above)
Comment (davidginzbourg) [reply tests/integration_tests/tests/agentless_tests/test_status_reporter.py]: Changed to contain 32 `a` letters

File: tests/integration_tests/tests/agentless_tests/test_status_reporter.py (hunk 4 above)
Comment (davidginzbourg) [reply tests/integration_tests/tests/agentless_tests/test_status_reporter.py]: Added underscores

File: tests/integration_tests/tests/agentless_tests/test_status_reporter.py (hunk 3 above)
Comment (davidginzbourg) [reply tests/integration_tests/tests/agentless_tests/test_status_reporter.py]: Done

**Pull request merged**